great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py

```diff
@@ -0,0 +1,151 @@
+            candidate_strings: a list of candidate date format strings that will REPLACE the default
+            additional_candidate_strings: a list of candidate date format strings that will SUPPLEMENT the default
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+
+        self._threshold = threshold
+        if candidate_strings is not None:
+            self._candidate_strings = set(candidate_strings)
+        else:
+            self._candidate_strings = self.CANDIDATE_STRINGS
+
+        if additional_candidate_strings is not None:
+            self._candidate_strings.update(additional_candidate_strings)
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        """
+        Check the percentage of values matching each string, and return the best fit, or None if no
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and
+        ptional details
+        """
+        validator: Validator = self.get_validator(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
+
+        nonnull_count = sum(
+            self.get_metrics(
+                batch_ids=batch_ids,
+                validator=validator,
+                metric_name="column_values.nonnull.count",
+                metric_domain_kwargs=self._metric_domain_kwargs,
+                domain=domain,
+                variables=variables,
+                parameters=parameters,
+            ).metric_values
+        )
+
+        format_string_success_ratios = dict()
+        for fmt_string in self._candidate_strings:
+            match_strftime_unexpected_count = sum(
+                self.get_metrics(
+                    batch_ids=batch_ids,
+                    validator=validator,
+                    metric_name="column_values.match_strftime_format.unexpected_count",
+                    metric_domain_kwargs=self._metric_domain_kwargs,
+                    metric_value_kwargs={"strftime_format": fmt_string},
```

Comment:
@alexsherstinsky - RE: `metric_value_kwargs`, I am struggling to understand a use case where a `ParameterBuilder` that computes more than one metric can utilize this argument. Take the above example where metric `column_values.match_strftime_format.unexpected_count` passes the candidate string to `metric_value_kwargs`, but metric `column_values.nonnull.count` does not. I understand that `metric_value_kwargs` could be provided to both metrics with the the candidate string provided as a supplement in the case of `match_strftime_format`, but how would the `metric_value_kwargs` be potentially be used? The [column_condition_partial](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/column_map_metrics/column_values_match_strftime_format.py) functions accept `**kwargs` that go unused. In the case of `metric_domain_kwargs` it is straightforward as the domain is shared between all computed metrics of the `ParameterBuilder`.

Reply:
@NathanFarmer I think that if we have a use for domain kwargs, we should include value kwargs in case various keys in a dictionary need to be passed to all metrics, and each metric will use the key(s) it needs needs from there.